projects/graphql-client/src/graphql-request.service.ts

```diff
@@ -155,7 +155,7 @@ export class GraphQlRequestService {
   ): Observable<TResponse> {
     return this.apollo
       .query<TResponse>({
-        query: gql(requestString),
+        query: gql(`query ${options?.name ?? ''} ${requestString}`),
```

Comment:
Will this work with current batching behavior? I would think it would either break or prevent batching - neither good. Work to push the name further down so multiple named queries can be grouped (you might want to use alias rather than a query name, for example).